### DISCUSSION TITLE
Come impostare il Calcolo Inverso in Zapier?

### DISCUSSION THREAD

**Discussion by angelarted** [fattureincloud/api#438] [Q&A] [answered]
Ciao,
mi manca solo un ultimo step per finalizzare la scrittura automatica delle fatture ad ogni acquisto fatto dalla piattaforma di e-learning: impostare il calcolo inverso.
Purtroppo non riesco a trovare nei vari campi che Zapier mi chiede di compilare, la voce corrispondente.
Anche nella documentazione di come scrivere una invoice non lo vedo ( [qui](https://developers.fattureincloud.it/docs/guides/invoice-creation) )
Potete gentilmente aiutarmi?

Grazie,

Simona

**Comment by valmoz** [comment, marked as answer]
Buongiorno @angelarted 
purtroppo questa funzionalità non è supportata lato API, quindi anche sulle action Zapier di Fatture in Cloud non è applicabile.

Credo che l'unico modo sia effettuare un calcolo con qualche altra action di Zapier (se possibile), magari utilizzando i gross prices in modo tale da far quadrare il totale da pagare con il totale degli items.
In alternativa, l'unica cosa che mi viene in mente è il [magic field](https://developers.fattureincloud.it/docs/guides/invoice-totals#tophat-the-magic-field), ma non credo che possa esserle utile in questo caso.

A disposizione
Mauro

**Reply by angelarted** [reply]
Grazie mille Mauro, 
sempre esaustivo e chiaro. 
Purtroppo avevo già provato con il magic field ma infatti non funziona (a parte che si può solo settare true o false)
Buona giornata!

**Reply by angelarted** [reply]
Ciao Ancora, 
il gross price non lo possiamo usare perché i nostri prodotti sono esenti da IVA.
Stavo guardando altri campi, posso chiederti cos'è il campo "Issued document global cassa taxable percentage - [data.global_cassa_taxable]"?
Perchè esiste già il campo "Issued document "Cassa previdenziale" percentual value - [data.cassa]" che è settato a 2, ed è questo quello che non dovrebbe essere incluso nel totale imponibile finale, per mimare il calcolo inverso nel nostro caso.
Forse questo campo di cui non capisco la funzione può influire sul calcolo? 
Altri suggerimenti per fare questo workaround?

Grazie infinite.
Simona

Edit: in realtà anche se cambio il valore nel campo, nell'output vedo sempre il valore 1, come mai???
Ho provato a inserire 2, 98, 0, ma l'output è sempre 1..
Ti allego screenshot:
![Schermata 2024-04-17 alle 12 20 24](https://github.com/fattureincloud/api/assets/5982081/8758ff3b-6587-4a6a-a978-e8e4cf701bd4)
![Schermata 2024-04-17 alle 12 22 36](https://github.com/fattureincloud/api/assets/5982081/ad528877-3ddd-497c-b1cf-e5d18f7a000a)
![Schermata 2024-04-17 alle 12 25 07](https://github.com/fattureincloud/api/assets/5982081/93654d7e-0135-4792-ab4d-2bef3292a40d)

**Reply by angelarted** [reply]
Nessuno mi risponde???? ahhhhhhh!! Sto impazzendo!!!!!  😥

**Reply by valmoz** [reply]
Buongiorno @angelarted 
non capisco, se i suoi prodotti sono esenti da IVA dovrebbe bastarle indicare i netti e impostare l'iva allo 0% per i prodotti come indicato dalla guida:
[Invoice Creation](https://developers.fattureincloud.it/docs/guides/invoice-creation#five-step-five-items-list)
Per il totale, essendo l'iva allo 0% dovrebbe bastarle sommare i netti, mi perdo qualcosa?

Comunque, le consiglio di provare prima le nostre API tramite la [API Reference](https://developers.fattureincloud.it/api-reference) e poi di replicare lo stesso comportamento su Zapier. In questo modo la risoluzione dei problemi dovrebbe essere più agevole.

Grazie
Mauro

**Reply by valmoz** [reply]
Buongiorno @angelarted è riuscita a risolvere?

**Reply by angelarted** [reply]
Ciao Mauro, in verità no, abbiamo tolto la voce dalla fattura. 
Avevo scritto una risposta tempo fa ma non la vedo più.
Provo a rispiegarle il paradosso dell'assenza di IVA ma di calcolo sbagliato in fattura.

Abbiamo prodotti esenti da IVA, ma con obbligo di un 2% a una cassa previdenziale e 2€ di bollo ogni transazione.

Questi costi aggiuntivi però non dovrebbero far parte dell'imponibile, cioè del TOTALE della fattura.
Con le fatture fatte a mano si usa il calcolo inverso e tutto torna magicamente al suo posto, il cliente riceve una fattura per il reale importo che ha pagato.
Ma con l'automazione non c'è verso di dire al flow di non tenere conto di quel 2% di cassa previdenziale e di quei 2 euro di bollo.

Come ti dicevo sopra, avevo pensato che il campo "global_cassa_taxable_percentage" potesse fare al nostro caso, ovvero se metto che nulla della percentuale dovuta alla cassa è tassabile, significa che non deve far parte dell'imponibile, quindi non va a finire nel totale della fattura.

Ma non è così.
E quindi (a meno di non fare delle simulazioni, usando i campi dell'IVA come se fossero CASSA - cosa che vorrei evitare) non c'è modo di risolvere il problema.

Grazie,
Simona

**Reply by valmoz** [reply]
Buongiorno @angelarted 
il mio consiglio in questo caso è quello di provare ad usare la API Reference (Get Issued Document) per recuperare un documento su cui avete applicato il Calcolo Inverso, e provare a replicare lo stesso comportamento tramite API Reference prima e tramite Zapier poi.

Mauro
